pkg/storage/etcd/etcd_helper.go

```diff
@@ -562,6 +613,9 @@ func (h *etcdHelper) GuaranteedUpdate(ctx context.Context, key string, ptrToType
 		if err != nil {
 			return err
 		}
+		if err := checkPreconditionsMatch(preconditions, obj); err != nil {
```

Comment:
Yeah, I think that's true. So OK.

On Thu, Mar 17, 2016 at 3:46 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/storage/etcd/etcd_helper.go
> <https://github.com/kubernetes/kubernetes/pull/22965#discussion_r56590353>
> :
>
> > @@ -562,6 +613,9 @@ func (h *etcdHelper) GuaranteedUpdate(ctx context.Context, key string, ptrToType
> >  		if err != nil {
> >  			return err
> >  		}
> > +		if err := checkPreconditionsMatch(preconditions, obj); err != nil {
>
> If the UID doesn't match, then the update should be aborted, because
> otherwise you are going to update an unexpected object.
>
> I think updating/deleting an object that doesn't match the UID
> precondition is always wrong. Do you have counterexamples?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/22965/files/97c67ef0c541c240d17e71af62f940ab792a0fe1#r56590353>
>